File: user-service/src/main/java/rs/edu/raf/model/FavouriteRecipient.java
package rs.edu.raf.model;

import jakarta.persistence.*;
import lombok.*;
import rs.edu.raf.model.user.Client;

@Entity
@Table
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class FavouriteRecipient {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne
    private Client client;
    private String recipientName;
    private String recipientAccountNumber;

    public FavouriteRecipient(Client client, String recipientName, String recipientAccountNumber) {
        this.client = client;
        this.recipientName = recipientName;
        this.recipientAccountNumber = recipientAccountNumber;
    }
}

File: user-service-kotlin/src/main/kotlin/org/example/userservicekotlin/model/FavouriteRecipient.kt
package org.example.userservicekotlin.model

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.ManyToOne
import org.example.userservicekotlin.model.user.Client

@Entity
class FavouriteRecipient(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long?,
    @ManyToOne val client: Client,
    val recipientName: String,
    val recipientAccountNumber: String
)
